crates/tendermint/src/block/height.rs

```diff
@@ -0,0 +1,65 @@
+use std::fmt::{self, Debug, Display};
+
+/// Block height for a particular chain (i.e. number of blocks created since
+/// the chain began)
+#[cfg_attr(feature = "serializers", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Default, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub struct Height(pub u64);
```

Comment:
I saw the Amino messages use `i64` for the block height... any particular reason why? Are negative block heights used for anything?

Reply:
It is an ongoing discussion (again). 
See: https://github.com/tendermint/tendermint/issues/2684 and https://github.com/tendermint/tendermint/issues/2692

Reply:
Subscribed to those.